src/libopensc/card-openpgp.c

```diff
@@ -2724,9 +2918,10 @@ pgp_build_extended_header_list(sc_card_t *card, sc_cardctl_openpgp_keystore_info
 	u8 *data = NULL;
 	size_t len = 0;
 	u8 *p = NULL;
-	u8 *components[] = {key_info->rsa.e, key_info->rsa.p, key_info->rsa.q, key_info->rsa.n};
-	size_t componentlens[] = {key_info->rsa.e_len, key_info->rsa.p_len,
-				  key_info->rsa.q_len, key_info->rsa.n_len};
+	u8 *components[] = {key_info->u.rsa.e, key_info->u.rsa.p,
+			    key_info->u.rsa.q, key_info->u.rsa.n};
+	size_t componentlens[] = {key_info->u.rsa.e_len, key_info->u.rsa.p_len,
+				  key_info->u.rsa.q_len, key_info->u.rsa.n_len};
```

Comment:
Shouldn't we initialize these values based on the key type?

Reply:
I thought about this as well. I just couldn't come up with a elegant way to do it. Do you have a suggestion? I don't like to put these values in a array separately...

Reply:
I don't think 19d4a34 is a masterpiece, but this is the only (easiest?) way I could think of to solve this early initialisation.